### PR TITLE
internal/charmstore: whitespace tokenize query text for autocomplete searches

### PR DESCRIPTION
**Pull request by mhilton**
Also ensure that all terms are matched in an auto-complete search.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1816/
Test PASSed.

File: elasticsearch/query.go [hunk 1]
```diff
@@ -60,15 +60,20 @@ func (m MatchQuery) MarshalJSON() ([]byte, error) {
 
 // MultiMatchQuery provides a query that matches on a number of fields.
 type MultiMatchQuery struct {
-	Query  string
-	Fields []string
+	Query              string
+	Fields             []string
+	MinimumShouldMatch string
```
Comment (rogpeppe) [inline elasticsearch/query.go:70]: It'd be nice to know what this means (either with a doc comment
or a link to the ES docs)

**Comment by makyo** [comment]
👍

File: internal/charmstore/search_test.go [hunk 2]
```diff
@@ -695,6 +695,15 @@ var searchTests = []struct {
 			searchEntities["wordpress"].entity,
 			searchEntities["wordpress-simple"].entity,
 		},
+	}, {
+		about: "autocomplete with spaces",
+		sp: SearchParams{
+			Text:         "wordpress simple",
```
Comment (rogpeppe) [inline internal/charmstore/search_test.go:701]: "simple wordpress" too? or is that something that will "obviously" work?

**Comment by rogpeppe** [comment]
LGTM assuming QA ok

File: elasticsearch/query.go (hunk 1 above)
Comment (mhilton) [inline elasticsearch/query.go:70]: done

File: internal/charmstore/search_test.go (hunk 2 above)
Comment (mhilton) [inline internal/charmstore/search_test.go:701]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1817/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1818/
Test PASSed.

**Pull request merged**